frontend/apps/core-app/src/components/RecordView/normaliseFieldValues.ts

```diff
@@ -24,39 +23,33 @@ export const normaliseFieldValues = (
 
     if (fieldType === FieldKind.SubForm) {
       if (!Array.isArray(cur.value)) return acc;
-      const values: NormalisedSubFormFieldValue[][] = cur.value.map(
-        (subRecord) => {
-          if (!Array.isArray(subRecord)) return [];
-          return subRecord.map((subFieldValue) => {
+      const item: NormalisedSubFormField = {
+        key: field.key,
+        header: field.name,
+        fieldType: FieldKind.SubForm,
+        columns:
+          field.fieldType.subForm?.fields.map((f) => ({
+            Header: f.name,
+            accessor: f.id,
+          })) ?? [],
+        data: cur.value.map((subRecord) => {
+          if (!Array.isArray(subRecord)) return {};
+          return subRecord.reduce<Record<string, string>>((a, c) => {
             const subField = field.fieldType.subForm?.fields.find(
-              (f) => f.id === subFieldValue.fieldId,
+              (f) => f.id === c.fieldId,
             );
             if (!subField) throw new Error('subField not found');
-            const subFieldType = getFieldKind(subField.fieldType);
-            if (subFieldType === FieldKind.SubForm)
-              throw new Error('subField is a subform');
+            if (getFieldKind(subField.fieldType) === FieldKind.SubForm)
+              throw new Error('Nested subforms are not supported');
```

Comment:
also not by the api, or just not handled in the frontend?

Reply:
I think the discussion concluded on, the api does support it but we should remove it at some point. The pwa does not support adding them.